interface/main/finder/dynamic_finder_ajax.php

```diff
@@ -93,30 +139,28 @@
 
 // Column-specific filtering.
 //
+$newWhere = array();
 for ($i = 0; $i < count($aColumns); ++$i) {
     $colname = $aColumns[$i];
     if (isset($_GET["bSearchable_$i"]) && $_GET["bSearchable_$i"] == "true" && $_GET["sSearch_$i"] != '') {
-        $where .= $where ? ' AND' : 'WHERE';
         $sSearch = add_escape_custom($_GET["sSearch_$i"]);
         if ($colname == 'name') {
-            if ($searchMethodInPatientList) { // exact search
-                $where .= " ( " .
-                    "lname LIKE '$sSearch' OR " .
-                    "fname LIKE '$sSearch' OR " .
-                    "mname LIKE '$sSearch' )";
-            } else {  // like search
-                $where .= " ( " .
-                    "lname LIKE '$sSearch%' OR " .
-                    "fname LIKE '$sSearch%' OR " .
-                    "mname LIKE '$sSearch%' )";
-            }
-        } elseif ($searchMethodInPatientList) {
-            $where .= " `" . escape_sql_column_name($colname, array('patient_data')) . "` LIKE '$sSearch'"; // exact search
+            $newWhere[] = " ( " .
+                "lname LIKE '".prepareMatchSearch($sSearch, "R", $searchMethodInPatientList)."' OR " .
+                "fname LIKE '".prepareMatchSearch($sSearch, "R", $searchMethodInPatientList)."' OR " .
+                "mname LIKE '".prepareMatchSearch($sSearch, "R", $searchMethodInPatientList)."' )";
         } else {
```

Comment:
Can't this else be removed and changed to:
```
} else if ($colname == "DOB") {
etc.
```

Reply:
Everything remained as before, just slightly improved the code.